### PR TITLE
Fix layering violations in brave_tooltips

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/17636

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by emerick** [comment]
Passed on Mac, Linux, and iOS. Restarting for Windows and Android.

**Comment by emerick** [comment]
Windows CI passed; restarting Android.

File: browser/brave_ads/tooltips/BUILD.gn [hunk 1]
```diff
@@ -24,4 +24,8 @@ source_set("tooltips") {
     "//chrome/browser/profiles:profile",
     "//ui/base",
   ]
+
+  if (is_win || is_linux || is_mac) {
```
Comment (simonhong) [inline browser/brave_ads/tooltips/BUILD.gn:28]: Adding view deps to `brave/browser/ui/BUILD.gn` would be much better like this.
```
if (brave_adaptive_captcha_enabled && toolkit_views) {
  deps += [ "//brave/browser/ui/views/brave_tooltips" ]
}
```

File: browser/ui/views/brave_tooltips/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import("//build/config/features.gni")
+import("//build/config/ui.gni")
+import("//testing/test.gni")
+import("//ui/base/ui_features.gni")
+
+assert(is_win || is_linux || is_mac)
```
Comment (simonhong) [inline browser/ui/views/brave_tooltips/BUILD.gn:11]: assert(brave_adaptive_captcha_enabled) ?

File: browser/brave_ads/tooltips/BUILD.gn (hunk 1 above)
Comment (emerick) [reply browser/brave_ads/tooltips/BUILD.gn]: `brave_adaptive_captcha` is already asserted in this file, so I don't think we need to add it there, but I did remove the `is_win || is_linux || is_mac` and added `toolkit_views` - thanks!

File: browser/ui/views/brave_tooltips/BUILD.gn (hunk 2 above)
Comment (emerick) [reply browser/ui/views/brave_tooltips/BUILD.gn:11]: My plan is that `brave_tooltips` will be more of a generic component. So while `brave_adaptive_captcha` does use tooltips, I don't necessarily want the tooltips component to know anything about `brave_adaptive_captcha` (or other components that use it). Does that make sense?

Comment (simonhong) [reply browser/ui/views/brave_tooltips/BUILD.gn:11]: Got it. I assumed adaptive_captcha is the only user of this.

**Review by simonhong — APPROVED** [review]

File: browser/ui/views/brave_tooltips/BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  ]
+
+  if (is_win) {
+    deps += [
+      "//ui/aura",
+      "//ui/wm/public",
+    ]
+  }
+
+  if (is_mac) {
```
Comment (aseren) [inline browser/ui/views/brave_tooltips/BUILD.gn:47]: seems like this frameworks are not needed

File: browser/ui/views/brave_tooltips/BUILD.gn [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  sources = [
+    "brave_tooltip_label_button.cc",
+    "brave_tooltip_label_button.h",
+    "brave_tooltip_popup.cc",
+    "brave_tooltip_popup.h",
+    "brave_tooltip_view.cc",
+    "brave_tooltip_view.h",
+  ]
+
+  if (is_win) {
```
Comment (aseren) [inline browser/ui/views/brave_tooltips/BUILD.gn:40]: are these deps needed?

File: browser/ui/views/brave_tooltips/BUILD.gn (hunk 3 above)
Comment (emerick) [reply browser/ui/views/brave_tooltips/BUILD.gn]: Thanks, I think you're right. Removed.

File: browser/ui/views/brave_tooltips/BUILD.gn (hunk 4 above)
Comment (emerick) [reply browser/ui/views/brave_tooltips/BUILD.gn]: It doesn't seem so, removed.

File: browser/brave_ads/tooltips/ads_tooltips_controller.cc [hunk 5]
```diff
@@ -9,7 +9,7 @@
 #include <utility>
 
 #include "brave/browser/ui/brave_tooltips/brave_tooltip_attributes.h"
-#include "brave/browser/ui/brave_tooltips/brave_tooltip_popup.h"
+#include "brave/browser/ui/views/brave_tooltips/brave_tooltip_popup.h"
```
Comment (aseren) [inline browser/brave_ads/tooltips/ads_tooltips_controller.cc:12]: As I understand we shouldn't include header from views here.
@simonhong @emerick WDYT?

Comment (simonhong) [reply browser/brave_ads/tooltips/ads_tooltips_controller.cc:12]: Oops, I missed this. Right. This also should be removed from this file.

File: browser/brave_ads/tooltips/BUILD.gn [hunk 6]
```diff
@@ -24,4 +25,8 @@ source_set("tooltips") {
     "//chrome/browser/profiles:profile",
     "//ui/base",
   ]
+
+  if (toolkit_views) {
```
Comment (simonhong) [inline browser/brave_ads/tooltips/BUILD.gn:29]: @emerick This deps including should be deleted from `//brave/browser/brave_ads/tooltips` target.
Instead, let's add this `//brave/browser/ui/views/brave_tooltips` target to `//brave/browser/ui`

File: browser/brave_ads/tooltips/ads_tooltips_controller.cc (hunk 5 above)
Comment (emerick) [reply browser/brave_ads/tooltips/ads_tooltips_controller.cc]: Thanks for pointing this out, fixed.

File: browser/brave_ads/tooltips/BUILD.gn (hunk 6 above)
Comment (emerick) [reply browser/brave_ads/tooltips/BUILD.gn]: Fixed.

**Comment by emerick** [comment]
All CI tests passed except Mac; will restart Mac now.

**Review by simonhong — APPROVED** [review]
++

**Review by aseren — APPROVED** [review]

**Comment by emerick** [comment]
CI passed on Mac; ready to merge.

**Pull request merged**